### PR TITLE
fix: enforce cookie ssl in production

### PR DESCRIPTION
**Pull request by alzafacon**
## Hey, I just made a Pull Request!

based on a codeql warning here is a fix to auth-backend

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/auth-backend/src/service/router.ts [hunk 1]
```diff
@@ -68,7 +68,8 @@ export async function createRouter(
   if (secret) {
     router.use(cookieParser(secret));
     // TODO: Configure the server-side session storage.  The default MemoryStore is not designed for production
-    router.use(session({ secret, saveUninitialized: false, resave: false }));
+    const enforceCookieSSL = process.env.NODE_ENV === 'production';
```
Comment (Rugvip) [inline plugins/auth-backend/src/service/router.ts:71]: Could we perhaps base it on the external URL of the backend instead? the `authUrl` in this code. I think that'd make sure we don't ruin things for people that really wanna run HTTP in production or that sorta thing :grin:

**Review by Rugvip — COMMENTED** [review]
Good call, thank you! 👍 

Just a minor nit on the detection method

File: plugins/auth-backend/src/service/router.ts (hunk 1 above)
Comment (alzafacon) [reply plugins/auth-backend/src/service/router.ts]: changed condition to use authUrl.startWith('https')

**Review by Rugvip — APPROVED** [review]
Nice, thanks! 👍

**Comment by Rugvip** [comment]
Ah, looks like this needs a rebase to fix the e2e test. There was a bad dep bump that is fix in more recent master

**Pull request merged**